File: tests/openmp_examples/memory_model/allocators.6.c
#pragma omp begin declare variant match(device={kind(host)})
#pragma omp end declare variant
#pragma omp begin declare variant match(device={kind(nohost)})
#pragma omp end declare variant
#pragma omp requires dynamic_allocators
#pragma omp declare target(calc)
#pragma omp declare target(cgroup_alloc)
#pragma omp target teams reduction(+:xbuf) thread_limit(256) allocate(omp_cgroup_mem_alloc:xbuf) num_teams(4)
#pragma omp parallel for
#pragma omp parallel for reduction(+:sum)
#pragma omp target
#pragma omp target
#pragma omp teams reduction(+:xbuf) thread_limit(256) allocate(cgroup_alloc:xbuf) num_teams(4)
#pragma omp parallel for
#pragma omp target
#pragma omp parallel for reduction(+:sum)

File: tests/openmp_examples/memory_model/allocators.6.f90
!$omp      declare target(calc)
!$omp     requires dynamic_allocators
!$omp     declare target(cgroup_alloc)
!$omp      target teams reduction(+:xbuf) thread_limit(N) allocate(omp_cgroup_mem_alloc:xbuf) num_teams(4)
!$omp         parallel do
!$omp     end target teams
!$omp     parallel do reduction(+:sum)
!$omp     target
!$omp     end target
!$omp      target
!$omp      teams reduction(+:xbuf) thread_limit(N) allocate(cgroup_alloc:xbuf) num_teams(4)
!$omp         parallel do
!$omp     end teams
!$omp     end target
!$omp     target
!$omp     end target
!$omp     parallel do reduction(+:sum)
